src/universal_transfer_operator/integrations/fivetran/connector/aws/s3.py

```diff
@@ -0,0 +1,292 @@
+    """
+
+    def __init__(
+        self,
+        connector_id: str | None,
+        service: str,
+        config: dict,
+        paused: bool = False,
+        pause_after_trial: bool = False,
+        sync_frequency: str | None = None,
+        daily_sync_time: str | None = None,
+        schedule_type: str | None = None,
+        trust_certificates: bool = False,
+        trust_fingerprints: bool = False,
+        run_setup_tests: bool = True,
+        connect_card_config: dict | None = None,
+    ):
+        self.connector_id = connector_id
+        self.service = service
+        self.config = config
+        self.paused = paused
+        self.pause_after_trial = pause_after_trial
+        self.sync_frequency = sync_frequency
+        self.daily_sync_time = daily_sync_time
+        self.schedule_type = schedule_type
+        self.trust_certificates = trust_certificates
+        self.trust_fingerprints = trust_fingerprints
+        self.run_setup_tests = run_setup_tests
+        self.connect_card_config = connect_card_config
+        super().__init__(
+            connector_id=self.connector_id,
+            service=self.service,
+            config=self.config,
+            paused=self.paused,
+            pause_after_trial=self.pause_after_trial,
+            sync_frequency=self.sync_frequency,
+            daily_sync_time=self.daily_sync_time,
+            schedule_type=self.schedule_type,
+            trust_certificates=self.trust_certificates,
+            trust_fingerprints=self.trust_fingerprints,
+            run_setup_tests=self.run_setup_tests,
+            connect_card_config=self.connect_card_config,
+        )
+
+    def to_dict(self) -> dict:
+        """
+        Convert options class to dict
+        """
+        return vars(self)
+
+    def _hook(self, conn_id: str, verify: Any, transfer_config_args: Any, s3_extra_args: Any) -> S3Hook:
+        """Return an instance of the database-specific Airflow hook."""
+        return S3Hook(
+            aws_conn_id=conn_id,
+            verify=verify,
+            transfer_config_args=transfer_config_args,
+            extra_args=s3_extra_args,
+        )
+
+    def map_airflow_connection_to_fivetran(
+        self,
+        source_dataset: Dataset,
+        destination_dataset: Dataset,
+        group_id: str,
+    ) -> dict[str, str]:
+        """
+        Maps the airflow connection details to fivetran config for destination
+
+        :param source_dataset: Source dataset
+        :param destination_dataset: Destination dataset
+        :param group_id: Group id in fivetran system
+        """
+        conn = BaseHook().get_connection(source_dataset.conn_id)  # type: ignore
+
+        access_key = conn.login
+        secret_key = conn.password
+
+        if not access_key:
+            raise ConnectionError(f"AWS_ACCESS_KEY_ID is not configured for {source_dataset.conn_id}")  # type: ignore
```

Comment:
what will happen if the access_key and secret_key are defined in env vars AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY?

Reply:
It's coming from airflow connections and we use `from airflow in a setting.configuration import conf` in settings.py The AWS_SECRET_ACCESS_KEY and AWS_ACCESS_KEY_ID is picked from env